python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,581 @@
+        # what we want for LSSTCam.
+        self.measurement.plugins["base_CircularApertureFlux"].radii = [
+            3.0,
+            4.5,
+            6.0,
+            9.0,
+            12.0,
+            17.0,
+            25.0,
+            35.0,
+            50.0,
+            70.0,
+        ]
+        lsst.meas.extensions.shapeHSM.configure_hsm(self.measurement)
+        self.measurement.plugins.names |= ["base_Jacobian", "base_FPPosition", "ext_photometryKron_KronFlux"]
+        self.measurement.plugins["base_Jacobian"].pixelScale = 0.2
+
+        # TODO: should make this the ApertureFlux default!
+        # Use a large aperture to be independent of seeing in calibration
+        self.measurement.plugins["base_CircularApertureFlux"].maxSincRadius = 12.0
+
+        # Only apply calibration fluxes, do not measure them.
+        self.normalized_calibration_flux.do_measure_ap_corr = False
+
+        # TODO: hopefully we can get rid of this, once Jim sorts out SDM.
+        self.post_calculations.plugins.names = ["base_LocalPhotoCalib", "base_LocalWcs"]
+        self.post_calculations.doReplaceWithNoise = False
+        for key in self.post_calculations.slots:
+            setattr(self.post_calculations.slots, key, None)
+
+
+class ReprocessVisitImageTask(pipeBase.PipelineTask):
+    """Use the visit-level calibrations to perform detection and measurement
+    on the single frame exposures and produce a "final" exposure and catalog.
+    """
+
+    ConfigClass = ReprocessVisitImageConfig
+    _DefaultName = "reprocessVisitImage"
+
+    def __init__(self, sources_schema=None, **kwargs):
+        super().__init__(**kwargs)
+
+        if sources_schema is None:
+            sources_schema = afwTable.SourceTable.makeMinimalSchema()
+
+        afwTable.CoordKey.addErrorFields(sources_schema)
+        self.makeSubtask("snap_combine")
+        self.makeSubtask("repair")
+        self.makeSubtask("detection", schema=sources_schema)
+        self.makeSubtask("sky_sources", schema=sources_schema)
+        self.makeSubtask("deblend", schema=sources_schema)
+        self.makeSubtask("measurement", schema=sources_schema)
+        self.makeSubtask("normalized_calibration_flux", schema=sources_schema)
+        self.makeSubtask("apply_aperture_correction", schema=sources_schema)
+        self.makeSubtask("catalog_calculation", schema=sources_schema)
+        self.makeSubtask("set_primary_flags", schema=sources_schema, isSingleFrame=True)
+        self.makeSubtask("post_calculations", schema=sources_schema)
+
+        sources_schema.addField(
+            "visit",
+            type="I",
+            doc="Visit this source appeared on.",
+        )
+        sources_schema.addField(
+            "detector",
+            type="U",
+            doc="Detector this source appeared on.",
+        )
+
+        # These fields will be propagated from finalizeCharacterization.
+        # TODO: get them from the finalized catalog instead, so the docstrings
```

Comment:
Is this done or need a ticket?

Reply:
It needs finalizeCharacterize to output a schema. But maybe I should just remove the TODO, since the docstrings don't propagate into parquet, as far as I know?

Reply:
I've removed this TODO and left it for future generations to sort out.